code/go/0chain.net/smartcontract/storagesc/test_blobber_reward.go

```
@@ -0,0 +1,183 @@
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	blockNumber := r.URL.Query().Get("block_number")
+	startBlockNumber := r.URL.Query().Get("start_block_number")
+	endBlockNumber := r.URL.Query().Get("end_block_number")
+
+	providerRewards := edb.GetRewardToProviders(blockNumber, startBlockNumber, endBlockNumber, spenum.FileDownloadReward.Int())
+	delegateRewards := edb.GetRewardsToDelegates(blockNumber, startBlockNumber, endBlockNumber, spenum.FileDownloadReward.Int())
+
+	result := map[string]interface{}{
+		"provider_rewards": providerRewards,
+		"delegate_rewards": delegateRewards,
+	}
+
+	common.Respond(w, r, result, nil)
+}
+
+func (srh *StorageRestHandler) getChallengeRewards(w http.ResponseWriter, r *http.Request) {
+	// read all data from challenge_rewards table and return
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	challengeID := r.URL.Query().Get("challenge_id")
+
+	blobberRewards, validatorRewards := edb.GetChallengeRewardsToProviders(challengeID)
+	blobberDelegateRewards, validatorDelegateRewards := edb.GetChallengeRewardsToDelegates(challengeID)
+
+	result := map[string]interface{}{
+		"blobber_rewards":            blobberRewards,
+		"validator_rewards":          validatorRewards,
+		"blobber_delegate_rewards":   blobberDelegateRewards,
+		"validator_delegate_rewards": validatorDelegateRewards,
+	}
+
+	common.Respond(w, r, result, nil)
+}
+
+func (srh *StorageRestHandler) getTotalChallengeRewards(w http.ResponseWriter, r *http.Request) {
+	// read all data from challenge_rewards table and return
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	allocationID := r.URL.Query().Get("allocation_id")
+
+	challenges, _ := edb.GetAllChallengesByAllocationID(allocationID)
```

Do not ignore errors